python/ray/data/dataset.py

```diff
@@ -287,6 +287,55 @@ def transform(block: Block) -> Iterable[Block]:
         blocks = compute.apply(transform, ray_remote_args, self._blocks)
         return Dataset(blocks, self._epoch, stats_builder.build(blocks))
 
+    def add_column(
```

Comment:
Hmm I'm not sure about the benefit-cost ratio: given we support pandas batch format, it should be trivial for users to just use map_batch?

Reply:
It's much more verbose when using map_batch (1 loc vs 5-6). I think this is quite confusing for users, who typically expect more streamlined utility functions.